web/terraform-checkpoints/replicate-config/main-foreach.tf

```diff
@@ -0,0 +1,319 @@
+# Terraform configuration to set up providers by version.
+terraform {
+  required_providers {
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = "~> 4.0"
+    }
+  }
+}
+
+# Configure the provider not to use the specified project for quota check.
+# This provider should only be used during project creation and initializing services.
+provider "google-beta" {
+  alias                 = "no_user_project_override"
+  user_project_override = false
+}
+
+# Configure the provider that uses the new project's quota.
+provider "google-beta" {
+  user_project_override = true
+}
+
+# Create new Google Cloud projects.
+resource "google_project" "default" {
+  provider = google-beta.no_user_project_override
+  name     = each.value
+  # Create a unique project ID for each project, with each ID starting with <PROJECT_ID>.
+  project_id = "<PROJECT_ID>-${each.key}"
+  # Required if you want to set up Authentication via Terraform
+  billing_account = "<YOUR_BILLING_ACCOUNT_ID>"
+
+  # Required for the projects to display in any list of Firebase projects.
+  labels = {
+    "firebase" = "enabled"
+  }
+
+  for_each = {
+    prod    = "<PROJECT_NAME_OF_PROD_PROJECT>"
+    staging = "<PROJECT_NAME_OF_STAGING_PROJECT>"
+  }
+}
+
+# Enable the required underlying Service Usage API.
+resource "google_project_service" "serviceusage" {
+  provider = google-beta.no_user_project_override
+  for_each = google_project.default
+
+  project = each.value.project_id
+  service = "serviceusage.googleapis.com"
+
+  # Don't disable the service if the resource block is removed by accident.
+  disable_on_destroy = false
+}
+
+# Enable the required underlying Firebase Management API.
+resource "google_project_service" "firebase" {
+  provider = google-beta.no_user_project_override
+  for_each = google_project.default
+
+  project = each.value.project_id
+  service = "firebase.googleapis.com"
+
+  # Don't disable the service if the resource block is removed by accident.
+  disable_on_destroy = false
+}
+
+# Enable Firebase services for the new project created above.
+resource "google_firebase_project" "default" {
+  provider = google-beta
+  for_each = google_project.default
+
+  project = each.value.project_id
+
+  # Wait until the required APIs are enabled.
+  depends_on = [
+    google_project_service.firebase,
+    google_project_service.serviceusage,
+  ]
+}
+
+# Create a Firebase Web App in the new project created above.
+resource "google_firebase_web_app" "default" {
+  provider = google-beta
+  for_each = google_firebase_project.default
+
+  project         = each.value.project
+  display_name    = "<DISPLAY_NAME_OF_YOUR_WEB_APP>"
+  deletion_policy = "DELETE"
+}
+
+# Uncomment below if you set up Firebase Authentication using Terraform in the previous step
```

Comment:
I would put these kinds of instructional comments in all-caps.

Reply:
Done.